File: c/chapitre_4/exo44Fils.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#define BUF 256
int main(int argc , char **argv)
{
    
    // Verification des arguments 
    if (argc != 2)
    {
        fprintf(stderr ,"Usage : %s <VariableEnvName> \n", argv[0]);
        exit(1);
    }
    
    char *val ;
    char message[BUF] ;
    char fd_read[10] , fd_write[10] ;

    // Lire la variable d'environement
    val = getenv(argv[1]);

    if (val == NULL )
    {
        fprintf(stderr,"Erreur var d'env\n");
        exit(0);
    }
    
    // Recuperation du tube
    sscanf(val , "%s:%s" , fd_read ,fd_write ) ;

    // Lecture du message
    close( atoi(fd_write) ) ;

    read( atoi(fd_read) , message , BUF ) ;

    printf("Fils : message recu => %s\n",message ) ;

    close( atoi(fd_read)) ;

    return (0);

}

File: python/chapitre_4/exo44Fils.py
import os
import sys

BUF = 256

def main():
    # Verification des arguments
    if len(sys.argv) != 2:
        print(f"Usage : {sys.argv[0]} <VariableEnvName>", file=sys.stderr)
        sys.exit(1)

    message = bytearray(BUF)

    # Lire la variable d'environement
    val = os.getenv(sys.argv[1])

    if val is None:
        print("Erreur var d'env", file=sys.stderr)
        sys.exit(0)

    # Recuperation du tube
    parts = val.split(':')
    fd_read  = parts[0]
    fd_write = parts[1]

    # Lecture du message
    os.close(int(fd_write))

    data = os.read(int(fd_read), BUF)
    print(f"Fils : message recu => {data.decode()}")

    os.close(int(fd_read))

    return 0

main()